fix: sd-teaser and sd-teaser-media a11y

smfonseca opened the pull request:
<!-- ## Title: Please consider adding the [skip chromatic] flag to the PR title in case you dont need chromatic testing your changes. -->
## Description:
This PR closes [1491](https://github.com/solid-design-system/solid/issues/1491), [1526](https://github.com/solid-design-system/solid/issues/1526) and  [1527](https://github.com/solid-design-system/solid/issues/1527)

__Note:__ I removed the alt text from the images in the templates, because they are decorative images and do not need alt text since no important information is conveyed by them.

## Definition of Reviewable:
<!-- *PR notes: Irrelevant elements should be removed.* -->
- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
<!-- *If this PR includes a bug fix, an E2E test is necessary to verify the change. If the fix is purely visual, ensuring it is captured within our chromatic screenshot tests is sufficient.* -->
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

MartaPintoTeixeira commented:
on the variants the order is wrong. "primary" should be the last on the list to match the samples. can you change this please?

MartaPintoTeixeira commented:
change the "Meta Slot" description to:
Use the “meta” slot to add additional content to the teaser. 

paulovareiro29 reviewed (approved):
LGTM

smfonseca commented:
> on the variants the order is wrong. "primary" should be the last on the list to match the samples. can you change this please?

Updated.

> change the "Meta Slot" description to:
Use the “meta” slot to add additional content to the teaser. 

Updated

MartaPintoTeixeira commented:
on teaser temlate instead of: "Unclickable Teaser With Button" change to "Unclickable Teaser with Button" please

MartaPintoTeixeira commented:
for teaser media template instead of "Teaser Media With Link" change to "Teaser Media with Link"

MartaPintoTeixeira commented:
for the Teaser Media Overrides sample template can you make it more similar to figma? 
"Gender" copy size is different in storybook

mariohamann commented on packages/docs/src/stories/templates/teaser.stories.ts:
```diff
@@ -91,32 +91,34 @@ export const ClickableTeaser = {
+                </sd-button>
+              </div>
+            </div>
+          </sd-teaser>
+        </a>
```
This is still no valid HTML as we have a button inside a link here.

mariohamann replied:
Ah. Just got aware of this ticket: https://github.com/solid-design-system/solid/issues/1541

So this will be tackled there, if I'm right?

smfonseca replied:
Yes, that's correct.

mariohamann commented on packages/docs/src/stories/templates/teaser-media.stories.ts:
```diff
@@ -10,19 +10,25 @@ export default {
-            src="./placeholders/images/coins.jpg"
-            alt="Close-up of hands stacking coins into small piles on a table, suggesting financial planning or saving."
-          />
+          <img class="w-full aspect-ratio" src="./placeholders/images/coins.jpg" alt="" />
```
Not sure, if these really count as decorative. Would love to hear Debbies thoughts on that: https://solid-design-system.github.io/solid/fix_sd-teaser-and-teaser-media-a11y/?path=/docs/templates-teaser--docs

koepferd replied:
As in the examples the images are not decorative but delivering emotions (e.g. Future is an attitude -> man with young children) I would actually add a meaningful alt attribute.
I suggest to add a small a11y hint though which says something like in case your used image is decorative keep it empty...

smfonseca commented:
@MartaPintoTeixeira 
> on teaser temlate instead of: "Unclickable Teaser With Button" change to "Unclickable Teaser with Button" please

Updated

> for teaser media template instead of "Teaser Media With Link" change to "Teaser Media with Link"

Updated

> on the 1st sample for teaser in figma we have the vertical variant.

Updated

> for the Teaser Media Overrides sample template can you make it more similar to figma?
"Gender" copy size is different in storybook

Synced directly on teams call. There's a ticket to make the [headline font size overridable](https://github.com/solid-design-system/solid/issues/1312). Decision now was to update Figma to resemble Storybook.
